### DISCUSSION TITLE
Combined config last_command_output and last_visisted_command_output

### DISCUSSION THREAD

**Discussion by bagge** [mikesmithgh/kitty-scrollback.nvim#274] [Q&A]
To reduce the number of key bindings for similar things I'll have to remember I have tried to achieve the following in kitty:
- If I run a command and then hit kitty_mod+g (my kitty-scrollback binding) it will start kitty-scrollback with the last command output.
- If I visit old command prompts with kitty_mod+z and kitty_mod+x and then hit kitty_mod+g I'd like to launch kitty-scrollback with the last visited command output.
(i.e. I am trying to avoid having to learn to use a separate key binding to open kitty-scrollback in the last_visited_command config)

Is there someway to achieve that? I have tried with some key sequence mappings in kitty.conf but that doesn't seem to work (makes it unable to just browse the old prompts). Was also contemplating to use modes but can't quite figure out how to get the intended behaviour. Or should I look for creating a special config in kitty-scrollback?

**Comment by mikesmithgh** [comment]
Hi @bagge 

That is not possible at the moment. You may be able to hack something together like this:

```lua
      require('kitty-scrollback').setup({
        ksb_builtin_last_cmd_output = function(kitty_data)
          local has_last_visited_cmd_output = vim
            .system({ kitty_data.kitty_path, '@', 'get-text', '--match=id:' .. kitty_data.window_id, '--extent', 'last_visited_cmd_output' })
            :wait().stdout
            :gsub('\n', '') ~= ''
          return {
            kitty_get_text = {
              extent = has_last_visited_cmd_output and 'last_visited_cmd_output' or 'last_cmd_output',
            },
          }
        end,
      })

```

This overrides the kitty-scrollback.nvim `ksb_builtin_last_cmd_output` builtin. If you have any last visited command output, then it will use that. Otherwise, it defaults to the last command output. This is a hacky solution but may work for your use case. 

If you don't mind two keymaps, something like this is the safer bet:
```
map kitty_mod+g kitty_scrollback_nvim --config ksb_builtin_last_cmd_output
map kitty_mod+l kitty_scrollback_nvim --config ksb_builtin_last_visited_cmd_output
```

Hope this helps.
